pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -240,6 +240,7 @@ func (cb *ClusterBuilder) buildDefaultPassthroughCluster() *cluster.Cluster {
 		ClusterDiscoveryType: &cluster.Cluster_Type{Type: cluster.Cluster_ORIGINAL_DST},
 		ConnectTimeout:       gogo.DurationToProtoDuration(cb.push.Mesh.ConnectTimeout),
 		LbPolicy:             cluster.Cluster_CLUSTER_PROVIDED,
+		ProtocolSelection:    cluster.Cluster_USE_DOWNSTREAM_PROTOCOL,
```

Comment:
This is the meat of the change, which fixes the actual bug. Everything else is testing

Reply:
Great diagnose!

Reply:
This is very interesting and I'm surprised that it has gone unnoticed until now.
With the "Use_configured_protocol" do we set it to HTTP 1/1?

Reply:
Yes, lack of http2_protocol_options == use h1

Reply:
The reason for it going unnoticed is that people that use grpc do not go thru this path, the port are explicitly marked as grpc.